### ISSUE TITLE
inWhere doesn't work with limit in query

### ISSUE THREAD

**Issue by saiger** [phalcon/cphalcon#10735]
example to reproduce:
```PHP
use Phalcon\DI,
    Phalcon\Mvc\Model,
    Phalcon\Db\Adapter\Pdo\MySQL as Connection,
    Phalcon\Mvc\Model\Manager as ModelsManager;


$di = new DI();

$connection = new Connection(array(
    "host"     => "localhost",
    "username" => "root",
    "password" => "",
    "dbname"   => "phalcon_test",
));

$modelsManager = new ModelsManager();
$modelsManager->setDi($di);
$di['db'] = $connection;
$di['modelsManager'] = $modelsManager;
$di['modelsMetadata'] = new Model\MetaData\Memory();

class Robots extends Model
{
    public $id;
}

$di->get('db')->execute('DROP TABLE IF EXISTS robots;');
$di->get('db')->execute('CREATE TABLE `robots` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;');

$robot = new Robots();
$robot->id = 1;
$robot->save();
$robot = new Robots();
$robot->id = 2;
$robot->save();

$robots = Robots::find();
echo 'found ' . $robots->count() . ' robots';
echo '<br>';

try {
Robots::query()->inWhere('id', [1, 2])->limit(1)->execute();
} catch (\Exception $exception) {
echo $exception->getMessage();
}

```
Got Output:
found 2 robots
SQLSTATE[HY093]: Invalid parameter number

**Comment by andresgutierrez** [comment]
Are you using Phalcon 2.0.6? I can't reproduce this issue.

Output:

```
SELECT IF(COUNT(*) > 0, 1, 0) FROM `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_NAME` = 'robots' AND `TABLE_SCHEMA` = DATABASE()
DESCRIBE `robots`
SELECT `robots`.`id`, `robots`.`name`, `robots`.`type`, `robots`.`year` FROM `robots` WHERE `robots`.`id` IN (:AP0, :AP1) LIMIT :AP0
Array
(
    [AP0] => 1
    [AP1] => 2
)
Array
(
    [AP0] => 1
)
```

**Comment by andresgutierrez** [comment]
I've added a fix for a different issue, maybe you want to try the latest version in the 2.0.x branch

**Comment by saiger** [comment]
Yep, it's tested against 2.0.5 and 2.0.6 Phalcon rev.

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
